how-to/hints-and-tips/docs/appassets-and-launch-external-process.md

```diff
@@ -0,0 +1,279 @@
+        "args": "a b c",
+        "mandatory": false
+    }
+]
+```
+
+The definition of an app asset and what each property means can be found here: [OpenFin.AppAssetInfo](https://developer.openfin.co/docs/javascript/stable/interfaces/OpenFin.AppAssetInfo.html).
+
+Both apps provide a source url, a version number for the zipped file, an alias which can be used for launching it, a target to indicate what file should be launched within the zip file once extracted and what command line args should be passed when calling it.
+
+The main difference between the two is that the first entry will cause the OpenFin application to fail if it can't be downloaded whereas the second entry says that the OpenFin app can continue loading if it fails to fetch the asset.
+
+#### App Assets via Code
+
+**One thing to note is that in the manifest approach above the RVM will fetch these assets on first launch of your manifest (or whenever the version of your asset changes)**. This might not be the behavior you want. You might want assets to be fetched after the user logs in or you might want to have dynamic app assets based on the currently logged in user.
+
+To define and fetch an asset via code you would do the following:
+
+```js
+try {
+    // this is hardcoded here but could have come from a rest api call
+  const appAsset: OpenFin.AppAssetInfo = {
+   src: "http://filesamples.com/exe.zip",
+   alias: "myApp",
+   version: "4.12.8",
+   target: "myApp.exe",
+   args: "a b c d"
+  };
+
+  await fin.System.downloadAsset(appAsset, (progress => {
+   const downloadedPercent = Math.floor((progress.downloadedBytes / progress.totalBytes) * 100);
+   console.log(`Downloaded ${downloadedPercent}% of app asset.`);
+  }));
+  console.log("App Asset is downloaded.");
+ } catch(err) {
+  console.error("There has been an error when trying to fetch the app asset.". err);
+ }
+```
+
+If you already have this version of the app asset then the RVM will not re-fetch it.
+
+#### Where do App Assets go?
+
+App assets are downloaded and extracted within your OpenFin application's folder. Using **myApp** as an example it would end up in the following location:
+
+> C:\Users\\{YOUR_USER_ID}\AppData\Local\OpenFin\apps\\{YOUR_APPLICATION}\assets\myApp\4.12.8
+
+#### Things to consider with App Assets
+
+There are a few things to consider with app assets:
+
+- Does your app need to write any files locally. If it does please note that the RVM will replace the versioned folder with a brand new versioned folder with the extracted zip contents. Locally written files will be removed. If you wish to persist files across installs then consider creating a directory one level above the exe e.g. C:\Users\\{YOUR_USER_ID}\AppData\Local\OpenFin\apps\\{YOUR_APPLICATION}\assets\myApp\4.12.8 would store it's data in C:\Users\\{YOUR_USER_ID}\AppData\Local\OpenFin\apps\\{YOUR_APPLICATION}\assets\myApp\data. Multiple versions could then check look for the data directory after an upgrade/install.
```

Comment:
Enclose the paths in backticks to make it more readable

Reply:
updated